handlers/product-switch-api/src/contributionToSupporterPlus.ts

```diff
@@ -56,56 +65,72 @@ export const switchToSupporterPlus = async (
 	};
 };
 
-const getContributionRefundAmount = (
-	zuoraResponse: ZuoraPreviewResponse,
+export const handleMissingRefundAmount = (
 	catalogInformation: CatalogInformation,
-	termStartDate: Date,
+	subscription: ZuoraSubscription,
+	currentDate: Date,
 ): number => {
-	const termStartDateDayOfMonth = termStartDate.getUTCDate();
-	const dayOfMonthToday = new Date().getUTCDate();
+	const ratePlan = getIfDefined(
+		subscription.ratePlans.find(
+			(ratePlan: RatePlan) =>
+				ratePlan.productRatePlanId ===
+				catalogInformation.contribution.productRatePlanId,
+		),
+		'No matching RatePlan found in Subscription,',
+	);
 
-	const invoice = getIfDefined(
-		zuoraResponse.previewResult?.invoices[0],
-		'No invoice found in the preview response',
+	const chargedThroughDate: Date = getIfDefined(
+		ratePlan.ratePlanCharges.find(
+			(ratePlanCharge: RatePlanCharge) =>
+				ratePlanCharge.productRatePlanChargeId ===
+				catalogInformation.contribution.chargeId,
+		)?.chargedThroughDate,
+		'No matching chargedThroughDate found in Subscription',
 	);
 
-	const contributionRefundAmount = invoice.invoiceItems.find(
-		(invoiceItem) =>
+	if (!(currentDate.toDateString() == chargedThroughDate.toDateString())) {
```

Comment:
```suggestion
	if (currentDate.toDateString() !== chargedThroughDate.toDateString()) {
```
I'm not sure if we really need it to be a dateString either just to compare it?  but JS is not my area of expertise